Source/Fuse.Scripting.JavaScript/Fuse.Scripting.JavaScript.unoproj

```diff
@@ -54,21 +54,5 @@
     "Duktape/duktape.c:File",
     "Duktape/duktape.h:File",
     "Duktape/duktape_helpers.h:File",
-
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSBase.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSContext.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSContextRef.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSExport.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSManagedValue.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSObjectRef.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSStringRef.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSStringRefCF.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSTypedArray.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSValue.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSValueRef.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JSVirtualMachine.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JavaScript.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/JavaScriptCore.h:File",
-    "JavaScriptCore/3rdparty/JavaScriptCore/WebKitAvailability.h:File"
```

Comment:
Uhhm, no... These *are* in use. For instance [here](https://github.com/fuse-open/fuselibs/blob/master/Source/Fuse.Scripting.JavaScript/JavaScriptCore/JavaScriptCore.uno#L6)...

Reply:
This copy of the headers that we have in our repo are not used. On iOS, we include system-installed headers of JavaScriptCore.

Currently, we are copying these headers when building for Android ([JavaScriptCore.cpp.uxl](https://github.com/fuse-open/fuselibs/blob/87d05c165098111909d73a58efbc4c654dc95f4c/Source/Fuse.Scripting.JavaScript/JavaScriptCore/3rdparty/JavaScriptCore.cpp.uxl)), where they're not needed because we use V8, and `-DUSE_JAVASCRIPTCORE` doesn't work on Android because we're missing the matching SO libs.

Reply:
We also support using JavaScriptCore on Android, and AFAIK we have some users who do that. This silently removes support for that.

Reply:
If `-DUSE_JAVASCRIPTCORE` doesn't work on Android anymore, then that's a regression. We definitely had people successfully using it.

Reply:
We pull in JSC [through gradle on Android](https://github.com/fuse-open/fuselibs/blob/87d05c165098111909d73a58efbc4c654dc95f4c/Source/Fuse.Scripting.JavaScript/JavaScriptCore/Context.uno#L8), so we don't need the SO in the tree.

Reply:
I see, I just tried it and got this output:
```
C:/Users/morten/AppData/Local/Android/Sdk/ndk-bundle/toolchains/llvm/prebuilt/windows-x86_64/lib/gcc/arm-linux-androideabi/4.9.x/../../../../arm-linux-androideabi/bin\ld: error: cannot find -ljsc
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2148: error: undefined reference to 'JSValueProtect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2154: error: undefined reference to 'JSValueToBoolean'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1520: error: undefined reference to 'JSObjectGetPrivate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1572: error: undefined reference to 'JSObjectGetPropertyAtIndex'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1672: error: undefined reference to 'JSObjectSetPropertyAtIndex'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2223: error: undefined reference to 'JSValueMakeNull'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2229: error: undefined reference to 'JSValueMakeNumber'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2217: error: undefined reference to 'JSValueMakeBoolean'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1688: error: undefined reference to 'JSObjectSetPrototype'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2161: error: undefined reference to 'JSValueToNumber'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1526: error: undefined reference to 'JSObjectGetProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2148: error: undefined reference to 'JSValueProtect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2235: error: undefined reference to 'JSValueMakeString'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1105: error: undefined reference to 'JSClassRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1105: error: undefined reference to 'JSClassRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1249: error: undefined reference to 'JSGlobalContextRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1330: error: undefined reference to 'JSContextGetGlobalObject'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:0: error: undefined reference to 'JSValueProtect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1105: error: undefined reference to 'JSClassRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1249: error: undefined reference to 'JSGlobalContextRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1255: error: undefined reference to 'JSEvaluateScript'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2229: error: undefined reference to 'JSValueMakeNumber'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2229: error: undefined reference to 'JSValueMakeNumber'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1336: error: undefined reference to 'JSGlobalContextCreate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1330: error: undefined reference to 'JSContextGetGlobalObject'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2142: error: undefined reference to 'JSValueIsObject'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1175: error: undefined reference to 'JSClassCreate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:0: error: undefined reference to 'kJSClassDefinitionEmpty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1162: error: undefined reference to 'JSClassCreate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:0: error: undefined reference to 'kJSClassDefinitionEmpty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2205: error: undefined reference to 'JSValueToStringCopy'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2223: error: undefined reference to 'JSValueMakeNull'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2229: error: undefined reference to 'JSValueMakeNumber'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2217: error: undefined reference to 'JSValueMakeBoolean'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1695: error: undefined reference to 'JSObjectMake'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1688: error: undefined reference to 'JSObjectSetPrototype'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1478: error: undefined reference to 'JSObjectCallAsConstructor'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2116: error: undefined reference to 'JSValueGetType'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2154: error: undefined reference to 'JSValueToBoolean'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1520: error: undefined reference to 'JSObjectGetPrivate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2133: error: undefined reference to 'JSValueIsInstanceOfConstructor'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1590: error: undefined reference to 'JSObjectHasProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2148: error: undefined reference to 'JSValueProtect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1493: error: undefined reference to 'JSObjectCallAsFunction'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1105: error: undefined reference to 'JSClassRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1249: error: undefined reference to 'JSGlobalContextRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1255: error: undefined reference to 'JSEvaluateScript'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1255: error: undefined reference to 'JSEvaluateScript'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1330: error: undefined reference to 'JSContextGetGlobalObject'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1841: error: undefined reference to 'JSStringCreateWithCharacters'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1508: error: undefined reference to 'JSObjectCopyPropertyNames'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1508: error: undefined reference to 'JSObjectCopyPropertyNames'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1520: error: undefined reference to 'JSObjectGetPrivate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1526: error: undefined reference to 'JSObjectGetProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1526: error: undefined reference to 'JSObjectGetProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1572: error: undefined reference to 'JSObjectGetPropertyAtIndex'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1572: error: undefined reference to 'JSObjectGetPropertyAtIndex'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1590: error: undefined reference to 'JSObjectHasProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1590: error: undefined reference to 'JSObjectHasProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1630: error: undefined reference to 'JSObjectSetProperty'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1672: error: undefined reference to 'JSObjectSetPropertyAtIndex'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1672: error: undefined reference to 'JSObjectSetPropertyAtIndex'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1688: error: undefined reference to 'JSObjectSetPrototype'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1760: error: undefined reference to 'JSPropertyNameArrayRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1760: error: undefined reference to 'JSPropertyNameArrayRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1766: error: undefined reference to 'JSPropertyNameArrayGetCount'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1766: error: undefined reference to 'JSPropertyNameArrayGetCount'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1772: error: undefined reference to 'JSPropertyNameArrayGetNameAtIndex' C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1772: error: undefined reference to 'JSPropertyNameArrayGetNameAtIndex' C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1826: error: undefined reference to 'JSStringGetLength'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1828: error: undefined reference to 'JSStringGetCharactersPtr'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1835: error: undefined reference to 'JSStringRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1835: error: undefined reference to 'JSStringRelease'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2211: error: undefined reference to 'JSValueUnprotect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2142: error: undefined reference to 'JSValueIsObject'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2142: error: undefined reference to 'JSValueIsObject'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2217: error: undefined reference to 'JSValueMakeBoolean'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2223: error: undefined reference to 'JSValueMakeNull'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2235: error: undefined reference to 'JSValueMakeString'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2235: error: undefined reference to 'JSValueMakeString'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2154: error: undefined reference to 'JSValueToBoolean'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2205: error: undefined reference to 'JSValueToStringCopy'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2205: error: undefined reference to 'JSValueToStringCopy'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2211: error: undefined reference to 'JSValueUnprotect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:2211: error: undefined reference to 'JSValueUnprotect'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1508: error: undefined reference to 'JSObjectCopyPropertyNames'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1766: error: undefined reference to 'JSPropertyNameArrayGetCount'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1772: error: undefined reference to 'JSPropertyNameArrayGetNameAtIndex' C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1114: error: undefined reference to 'JSObjectGetPrivate'
C:\foo\build\Android\Debug\app\src\main\jni/Fuse.Scripting.JavaScriptCore.g.cpp:1151: error: undefined reference to 'JSValueIsObject'
```

Reply:
But I should probably change condition in the UXL file to say `ANDROID && USE_JAVASCRIPTCORE` instead, to avoid copying these headers when JavaScriptCore isn't used.

Reply:
Fixed now.

Reply:
The latest commit in combination with https://github.com/fuse-open/uno/pull/171 fixes the JSC linking problem too.